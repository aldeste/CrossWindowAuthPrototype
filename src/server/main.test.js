const supertest = require("supertest");
const server = require("./main").default;

// jest.unmock("supertest").unmock("express");

beforeEach(() => {
  console.log = () => jest.fn();
});

afterEach(() => {
  server.close();
});

describe("Express server starts server", () => {
  it("gives a response when accessing '/'", async () => {
    const response = await supertest(server).get("/");
    expect(response.status).not.toBe(200);
  });
});

// describe("Accessing /connect yields response", () => {
//   it("recieves a response", async () => {
//     const response = await supertest(server).get("/connect");
//     expect(response.status).toBe(200);
//   });
// });

describe("Accessing /graphql", () => {
  it("yields response", async () => {
    const response = await supertest(server).get("/graphql");
    expect(response.header.location).toBe("/");
  });
});

describe("Queries", () => {
  beforeEach(async () => {
    await require("../database").initializeDatabase();
  });

  it("queries with graphql", async () => {
    const response = await supertest(server).get(
      "/?query={viewer(personId:4){id,name}}"
    );
    expect(response.text.data).toMatchSnapshot();
    expect(response.status).toBe(200);
  });

  it("runs timer diagnostics", async () => {
    const response = await supertest(server).get(
      "/?query={viewer(personId:4){id,name}}"
    );
    expect(JSON.parse(response.text).extensions.timeTaken).toBeDefined();
  });
});
