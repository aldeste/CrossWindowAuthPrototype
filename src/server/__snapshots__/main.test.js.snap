// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Express server starts server gives a response when accessing '/' 1`] = `400`;
